### DISCUSSION TITLE
Best modding tools for NZP?

### DISCUSSION THREAD

**Discussion by Dorr21** [nzp-team/nzportable#374] [General]
I have recently gotten back into playing NZP, and I was thinking of starting to map and mod for it. I already understand the best tool for mapping (Trenchbroom), but I am a little interested in other aspects of modding, the issue being that I don't know the best tools for aforementioned other aspects, those being:

- Modeling (I got a copy of Milkshape 3D recently, but I can't find the function to import and export Quake 1 MDL files)
- Texture conversion and creation (Including texture unwrapping for models)
- QuakeC editing and compilation
- Audio editing

Thank you so much in advance for your response, and I hope that I can one day provide for the community.

**Comment by MotoLegacy** [comment]
Hey there! Our README actually explains a lot of this, pointing you in the direction for QuakeC and compilation, modeling tools and plugins, etc. Be sure to [check that out!](https://github.com/nzp-team/nzportable/tree/main/README.md)

Here are the answers it won't provide:

* External Textures that aren't baked into the MDL (custom palette, unique color range) are .PCX format. I personally use GIMP to create those. Using the `Image->Mode->Indexed` menu options and converting a texture to 255 colors (default option) and then exporting to .PCX is all you need. The file is placed alongside the .MDL, check some of our weapons shipped with the game for reference :) 

* Audio files are unsigned 8-bit PCM .WAVs. They are easily exportable from "Audacity" (or forks). The `rate` is a very important factor. The lower the rate, the less RAM the file will take up, but the less clear it will sound. NZP usually uses either 8000Hz or 11000Hz depending on the context. 

Please reply if you have any more questions, thanks!

**Reply by Dorr21** [reply]
Oh, sorry about that. Thank you very much for your response, I hope I will be able to learn more about the structure of NZP and quake-based games in general, and maybe even contribute in the future. I wish you a good day.
